app/components/SearchResults.tsx

```diff
@@ -2,30 +2,44 @@ import { Link } from "@remix-run/react"
 import React from "react"
 import type JSXNode from "~/types/JSXNode"
 import SanitizedTextSpan from "./SanitizedTextSpan"
+import { SearchResultEnum } from "~/routes/search/searchResultEnum"
 
 interface SearchResultsProps {
-  data: Record<string, any[]>
+  data: any[]
   text: string
   pageNumber: string | undefined
+  searchAttribute: string | null
 }
 
-function displayEntries(
+function getSearchResults(
   pageNumber: string,
   text: string,
-  data: Record<string, any[]>
+  data: any[],
+  attribute: string | null
 ) {
-  if (data.entries === undefined || data.entries.length === 0) {
+  switch (attribute) {
+    case SearchResultEnum.HEADWORD:
+      return displayEntries(text, data)
+    case SearchResultEnum.MEANING:
+      return displayMeanings(text, data)
```

Comment:
Rather than calling `displayEntries` or `displayMeanings`, let's just factor these two functions out into first-class React components, e.g. `<SearchResultEntries ... />` and `<SearchResultMeanings ... />`.